### PR TITLE
Docs: Update README: Tech Stack; Links for Vision, MVP, Roadmap; Clarify terminology

### PR DESCRIPTION
**Pull request by allella**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

Closes: nothing

- Change Sequelize to TypeORM
- Vision statement link to the Wiki vision page
- Reword terminology section based on suggestions from confusioned Github members
- Update post-MVP / Roadmap to link to the GitHub Roadmap tag
- Move the API spec below the Tech Stack and Development Setup since seeing the API spec requires these earlier steps
- Change NYC to New York City for clarity
- Add credit DBeaver for the ER Diagram image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,11 +126,7 @@ npm run test:watch
 The [MVP user stories are shown in the MVP Project](https://github.com/freeCodeCamp/chapter/projects/1) kanban / cards and as [issues marked with "MVP"](https://github.com/freeCodeCamp/chapter/labels/MVP).
 
 ### Post-MVP
-We are maintaining a list of [post-MVP user stories](https://github.com/freeCodeCamp/chapter/issues/84).
-
-## Roadmap
-
-The on-going [project Roadmap conversation](https://github.com/freeCodeCamp/chapter/issues/47) is regularly updated to reflect the overall progress and for higher-level discussions.
+We are maintaining a list of post-MVP user stories using the ["Roadmap" tag](https://github.com/freeCodeCamp/chapter/labels/Roadmap).
```
Comment (vaibhavsingh97) [inline README.md]: as currently no issues are tagged as `Roadmap` so shouldn't we use the future tense here?

Comment (allella) [reply README.md]: Thanks @vaibhavsingh97 . Quincy purposely closed the to-do's on the Roadmap. What do you think about linking to the [(open or closed) + Roadmap tag](https://github.com/freeCodeCamp/chapter/issues?utf8=%E2%9C%93&q=is%3Aopen+or+is%3Aclosed+label%3ARoadmap+)? 

We can remove the Roadmap tag when items are closed and not on the roadmap.  Or, perhaps by the time we get into the roadmap Quincy will reopen them and we can link to open + Roadmap.

Comment (allella) [reply README.md]: I've pushed another commit with a better Roadmap tags link.

Comment (vaibhavsingh97) [reply README.md]: Thanks @allella it's perfect now 💯

**Pull request merged**